File: sams-backend/src/main/java/com/sams/entity/DeviceCode.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_device_code", schema = "asset")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DeviceCode {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "device_code_id")
    private Long deviceCodeId;

    @Column(name = "device_code")
    private String deviceCode;

    @Column(name = "device_concept")
    private String deviceConcept;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "org_id")
    private Long orgId;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}

File: SAMSUI/src/app/Model/master/DeviceCode.ts
import { BaseModel } from "../base/baseModel";

export class DeviceCodeModel extends BaseModel{
    public deviceCodeId ?: number;

    public deviceCode ?: string;

    public deviceConcept ?: string;

    public active ?: boolean;

    public createdBy ?: string;

    public createdDt ?: Date;

    public updatedBy ?: string;

    public updatedDt ?: Date;

     public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public createdDtDisp ?: string;

    public orgId ?: number;
}